Removal of Salary Structure Field on Employee Doctype

Talleyrand333 opened the pull request:
## Is this a Feature, Chore or Bug?
- [] Feature
- [*] Chore
- [] Bug


## Clearly and concisely describe the feature, chore or bug.
-Removal of Salary Structure Field on Employee Doctype
- Base and Salary Structure made mandatory on Job Offer Doctype


## Analysis and design (optional)
Analyze and attach the design documentation


## Solution description
Describe your code changes in detail for reviewers.
- I created a patch to delete the custom field from the database
- I customized the Job Offer form to make the base and salary structure fields mandatory

## Is there a business logic within a doctype?
    - [] Yes
    - [] No


## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.


## Areas affected and ensured
List out the areas affected by your code changes.
- Employee Doctype
- Job Offer Doctype


## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.
No , There was no reference found to the deleted field


## Did you test with the following dataset?
- [] Existing Data
- [*] New Data

## Was child table created?
No
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [*] Yes
    - [] No
        If yes, did you write a delete patch?
          Yes

## Is patch required?
- [*] Yes
- [] No
    ## Was the patch test?
    Yes



## Which browser(s) did you use for testing?
  - [*] Chrome
  - [] Safari
  - [] Firefox

SalShai committed:
Fetch the value for component

SalShai committed:
Update the table when change in field

pjamsheer committed:
Merge pull request #1566 from ONE-F-M/183898797

Fix: Salary structure Assignment

SalShai committed:
Optimize the Checkin Hook Events

pjamsheer committed:
Merge pull request #1567 from ONE-F-M/183933271

Optimize Employee Checkin Hook Events

Commit:
Removed Custom fields

mymi14s commented on one_fm/api/doc_events.py:
```diff
@@ -40,16 +39,14 @@ def employee_checkin_validate(doc, method):
+			if curr_shift:
 				start_date = (curr_shift.start_date).strftime("%Y-%m-%d")
 				existing_perm = frappe.db.exists("Shift Permission", {"date": start_date, "employee": doc.employee, "permission_type": perm_map[doc.log_type], "workflow_state": "Approved"})
-				doc.shift_assignment = last_shift[0].name
```
What business has shift assignment in this story?

Talleyrand333 replied:
I'm not sure how it happened but It seems other commits are also added. I did not make those changes in my branch.

mymi14s replied:
@Talleyrand333 keep your branches upto date.

Talleyrand333 replied:
@mymi14s that seems to be the issue, I did not fetch the latest master-14 branch. I will do that.

pjamsheer commented:
@Talleyrand333 `Removal of Salary Structure Field on Employee Doctype` need to remove the field from the code base (custom/employee.json(

Commit:
removed field from .json

Talleyrand333 commented:
@pjamsheer I added a patch [here](https://github.com/ONE-F-M/One-FM/blob/183851512/one_fm/patches/v14_0/remove_salary_structure.py). I have also removed the field from the employee.json file.
